### ISSUE TITLE
NOAUTH Authentication required

### ISSUE THREAD

**Issue by btmurrell** [NodeRedis/node_redis#989]
This is the error you get if you do not authenticate with node_redis.  According to the documentation, the options passed to `createClient()` may include a **password** property.  The doc says:

> password: null; If set, client will run redis auth command on connect. Alias auth_pass

With release 2.4.2, this definitely does *not* happen. If your redis server requires authorization, you must call `redisClient.auth(password)` as the first call after `createClient()`.

**To the node_redis team:** Your top-level README in the `createClient()` **options** section needs to get in sync with the code.  If you plan to implement automatic `auth(password)` invocation if options.password is provided, that's the bug.  If not, it should be removed from the README.

**Comment by BridgeAR** [comment]
I'm sorry, I made a mistake by merging something to early. It'll work from 2.5 on. 

You can use `auth_pass` in 2.4 instead (password is just an alias in 2.5 that sounds nicer).

Sorry for the inconvenience :/

**Comment by BridgeAR** [comment]
[2.5.0-1](https://github.com/NodeRedis/node_redis/releases/tag/2.5.0-1) got just released and implements the password option.

**Comment by btmurrell** [comment]
right on. thanks @BridgeAR.  i know this is just a nit, but my intention for logging the issue was so others could get a hit on 'NOAUTH'... i found many other explanations for the error, but i traced my occurrance to here.

**Comment by BridgeAR** [comment]
I'm uncertain what you mean. All errors are either returned in a specified callback or emitted to the error listener.

If you wanted to indicate that the docs should be changed too, that's what I already did (even though I might switch the alias still).

**Comment by btmurrell** [comment]
@BridgeAR , sorry. I just meant that with node_redis, I got the NOAUTH error (which i found comes directly from Redis server response).  I searched the web for solutions, but did not find one for node_redis not making the auth() call. So, for others encountering this problem, I just wanted to have google land on this issue... just documenting my experience and solution in what I assume is the best place for it.  Thanks for getting your fix rolled in so quickly.
